controller/src/main/java/io/pravega/controller/store/stream/StreamMetadataStore.java

```diff
@@ -287,6 +289,67 @@
                                                                                            final OperationContext context,
                                                                                            final Executor executor);
 
+
+    /**
+     * Updates the subscribers metadata for an existing stream.
```

Comment:
nit: create. 
also store can expose AddSubscriber and RemoveSubscriber apis. 
internally the subscriber record can be created at the time of stream creation and is updated whenver a new subscriber is added/removed by doing a get and set.
and this can fail with write conflict to the caller may need to handle and retry the operation.
we follow optimistic concurrency.. the assumption here is that subscriber registration is a rare occurrence and there will not be many conflicts.